lib/ecto/postgres/pg_framerate_migrations.ex

```diff
@@ -91,8 +92,11 @@ defpgmodule Vtc.Ecto.Postgres.PgFramerate.Migrations do
   def create_all do
     :ok = create_type_framerate_tags()
     :ok = create_type_framerate()
+
     :ok = create_function_schemas()
 
+    :ok = create_func_is_ntsc()
+
     :ok
```

Comment:
I would suggest restructuring this (and other similar functions) so you can capture the real underlying errors, i.e. 

```
def create_all do
    with :ok <- create_type_framerate_tags(),
             :ok <- create_type_framerate(),
             :ok <- create_function_schemas(),
             :ok <- create_func_is_ntsc() do
                  :ok 
     end
end
```

and then the helper functions would look something like 

```
 def create_type_framerate_tags do
      Migration.execute("""
        DO $$ BEGIN
          CREATE TYPE framerate_tags AS ENUM (
            'drop',
            'non_drop'
          );
          EXCEPTION WHEN duplicate_object
            THEN null;
        END $$;
      """)
  end
```

this ensures that if any of the helper functions fail, we'll pass through the actual error from `Migration.execute/2` instead of returning something like `(MatchError) no match of right hand side value` which is less helpful

Reply:
That would normally be how I would approach something like this, but the Migration.execute's return `:ok` is purely aesthetic. It always raises on errors. My guess is that it's to avoid the return type being `term`. From `Ecto.Migration.Runner`:

```elixir
  def execute(command) do
    reply =
      Agent.get_and_update(runner(), fn
        %{command: nil} = state ->
          {:ok, %{state | subcommands: [], commands: [command|state.commands]}}
        %{command: _} = state ->
          {:error, %{state | command: nil}}
      end)

    case reply do
      :ok ->
        :ok
      :error ->
        raise Ecto.MigrationError, "cannot execute nested commands"
    end
  end
```

I'm not sure why I'm matching on it. It just feels wrong to call the function without checking, but maybe the answer is to not match to avoid confusion for readers?

Reply:
Made that change in 462281841ad8396276633f67d28fb548aac2f1b7

Does that seem sensible to you?

Reply:
yes, that works!